pkg/client/annotate_manifest.go

```diff
@@ -0,0 +1,86 @@
+package client
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/google/go-containerregistry/pkg/name"
+
+	"github.com/buildpacks/pack/pkg/image"
+)
+
+type ManifestAnnotateOptions struct {
+	// Image index we want to update
+	IndexRepoName string
+
+	// Name of image we wish to update into the image index
```

Comment:
Currently, if it is not a digest we try to find the latest digest for the given image reference based on your host platform I think. for example:

`pack manifest annotate foo/my-index foo/my-manifest --os linux`

tries to look for foo/my-manifest:latest in the registry filtering by my current os and then it will determine the digest if something was found, I do not thing this is bad, but it makes the operation slower because we do that remote search. 

I think, we could simplify and only accept the digest, because at the end of the day, we are only updating the index json file and there is were we need to look for.